pkg/rhoda/rhoda_test.go

```diff
@@ -18,165 +22,379 @@ import (
 
+					By("checking Provider Acct deleted")
+					Eventually(func() bool {
+						err := client.Get(context.Background(), k8sClient.ObjectKeyFromObject(&inventory), &dbaasv1alpha1.DBaaSInventory{})
+						if err != nil && errors.IsNotFound(err) {
+							fmt.Println("Deleted, no provider acct found")
+							return true
+						}
+						return false
+					}, 60*time.Second, 5*time.Second).Should(BeTrue())
+					//}
+				})
+
+				fmt.Println("Creating secret for : " + provider.ProviderName)
 				//create secret
 				secret := core.Secret{
 					TypeMeta: meta.TypeMeta{
 						Kind:       "Secret",
 						APIVersion: "v1",
 					},
 					ObjectMeta: meta.ObjectMeta{
-						Name:      "dbaas-secret-e2e-" + providerName,
+						Name:      provider.SecretName,
 						Namespace: namespace,
 					},
-					Data: secretData,
+					Data: provider.SecretData,
 				}
-				fmt.Println(secret)
-				//_, err = clientset.CoreV1().Secrets("openshift-dbaas-operator").Create(context.TODO(), &secret, meta.CreateOptions{})
-				//Expect(err).NotTo(HaveOccurred())
+				_, err = clientset.CoreV1().Secrets("openshift-dbaas-operator").Create(context.TODO(), &secret, meta.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
 
-				//add to array
-				providers = append(providers, rhoda.ProviderAccount{ProviderName: providerName, SecretName: "dbaas-secret-e2e-" + providerName, SecretData: secretData})
-			}
-		})
+				//create inventory
+				fmt.Println("Creating inventory for : " + provider.ProviderName)
+				inventory := dbaasv1alpha1.DBaaSInventory{
+					TypeMeta: meta.TypeMeta{
+						Kind:       "DBaaSInventory",
+						APIVersion: "dbaas.redhat.com/v1alpha1",
+					},
+					ObjectMeta: meta.ObjectMeta{
+						Name:      "provider-acct-test-e2e-" + provider.ProviderName,
+						Namespace: namespace,
+						Labels:    map[string]string{"related-to": "dbaas-operator", "type": "dbaas-vendor-service"},
+					},
+					Spec: dbaasv1alpha1.DBaaSOperatorInventorySpec{
+						ProviderRef: dbaasv1alpha1.NamespacedName{
+							Namespace: namespace,
+							Name:      string(provider.SecretData["providerType"]),
+						},
+						DBaaSInventorySpec: dbaasv1alpha1.DBaaSInventorySpec{
+							CredentialsRef: &dbaasv1alpha1.NamespacedName{
+								Namespace: namespace,
+								Name:      provider.SecretName,
+							},
+						},
+					},
+				}
+				err = client.Create(context.Background(), &inventory)
+				Expect(err).NotTo(HaveOccurred())
 
-		Context("Create Inventory", func() {
-			It("Create inventory using the map", func() {
-				for _, value := range providers {
-					fmt.Println(value)
+				//sleep timer to make sure things run in parallel
+				//fmt.Printf("Current Unix Time: %v\n", time.Now())
+				//time.Sleep(30 * time.Second)
+				//fmt.Printf("Current Unix Time: %v\n", time.Now())
 
-					scheme := runtime.NewScheme()
-					err := dbaasv1alpha1.AddToScheme(scheme)
+				//Check inventories status
+				inventory = dbaasv1alpha1.DBaaSInventory{}
```

Comment:
Rename this so you don't reuse the same variable name as above, and move it inside the Eventually block